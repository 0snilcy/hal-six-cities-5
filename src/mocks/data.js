// import { getDate } from '../utils/utils';
// const getRandomNumber = (number) => getCountRandom(0, number);

const getRandomBoolean = () => Boolean(Math.round(Math.random()));

export const getCountRandom = (min, max) =>
  Math.floor(getCountRandomDecimal(max, min));

const getCountRandomDecimal = (min, max) =>
  Number((Math.random() * (max - min) + min).toFixed(1));

export const city = [
  'Paris',
  'Cologne',
  'Brussels',
  'Amsterdam',
  'Hamburg',
  'Dusseldorf',
];

const getRatingPercent = (value) => {
  return value / 0.05;
};

export const getData = () => ({
  bedrooms: getCountRandom(1, 4),
  // city: {
  //   location: {
  //     latitude: 52.370216,
  //     longitude: 4.895168,
  //     zoom: 10,
  //   },
  //   name: 'Amsterdam',
  // },
  description: [
    `Very recommended: comfortable location, incredibly friendly staff, very tasty breakfasts, cleanliness, comfort. A decent hotel! You're great, so keep it! Thanks for the hospitality, be sure to return to you.`,
    `Great hotel. Good location. Everything is clean and cozy. A variety of breakfasts. I thank the hotel staff. I would like to note that everyone is very friendly and polite, so an atmosphere of good rest and excellent mood is created.`,
    `Despite its inexpensive price the hotel is very decent. Always friendly staff, willingly responded to any requests. I was pleased that before breakfast the temperature was measured and gloves and masks were given out. The location of the hotel is very comfortable, in the center you can see everything and close to the station.`,
  ][getCountRandom(0, 2)],
  goods: [
    `Wi-Fi`,
    `Washing machine`,
    `Towels`,
    `Heating`,
    `Coffee machine`,
    `Baby seat`,
    `Kitchen`,
    `Dishwasher`,
    `Cabel TV`,
    `Fridge`,
  ].slice(getCountRandom(0, 4), getCountRandom(4, 9)),
  // host: {
  //   avatar_url: 'img/1.png',
  //   id: 3,
  //   is_pro: true,
  //   name: 'Angelina',
  // },
  // id: 1,
  images: [
    'img/apartment-01.jpg',
    'img/apartment-02.jpg',
    'img/apartment-03.jpg',
  ][getCountRandom(0, 2)],
  isFavorite: getRandomBoolean(),
  isPremium: getRandomBoolean(),
  // location: {
  //   latitude: 52.35514938496378,
  //   longitude: 4.673877537499948,
  //   zoom: 8,
  // },
  adults: getCountRandom(1, 5),
  previewImage: [
    `img/apartment-small-03.jpg`,
    `img/room-small.jpg`,
    `img/apartment-small-04.jpg`,
  ][getCountRandom(0, 2)],
  price: Math.round(getCountRandom(40, 200) / 10) * 10,
  rating: getRatingPercent(getCountRandomDecimal(3, 5)),
  title: [
    `Beautiful & luxurious apartment at great location`,
    `Wood and stone place`,
    `Nice, cozy, warm big bed apartment`,
    `Canal View Prinsengracht`,
  ][getCountRandom(0, 3)],
  type: [`apartments`, `private room`, `studio`, `loft`][getCountRandom(0, 3)],
  tariff: [`night`, `light`][getCountRandom(0, 1)],
});

// export const getOffer = () => ({
//   picture: getCountRandom(1, 4),
//   isPremium: getRandomBoolean(),
//   name: [
//     `Beautiful & luxurious apartment at great location`,
//     `Wood and stone place`,
//     `Nice, cozy, warm big bed apartment`,
//     `Canal View Prinsengracht`,
//   ][getCountRandom(0, 3)],
//   ratingValue: getCountRandomDecimal(2, 5),
//   ratingView: getCountRandom(20, 100),
//   classRoom: [`apartments`, `private room`, `studio`, `loft`][
//     getCountRandom(0, 3)
//   ],
//   bedrooms: getCountRandom(1, 4),
//   adults: getCountRandom(1, 5),
//   price: Math.round(getCountRandom(40, 200) / 10) * 10,
//   tariff: [`night`, `light`][getCountRandom(0, 1)],
//   conveniences: [
//     `Wi-Fi`,
//     `Washing machine`,
//     `Towels`,
//     `Heating`,
//     `Coffee machine`,
//     `Baby seat`,
//     `Kitchen`,
//     `Dishwasher`,
//     `Cabel TV`,
//     `Fridge`,
//   ].slice(getCountRandom(0, 4), getCountRandom(4, 9)),
//   reviews: new Array(3).fill('').map(reviewsDate),
// });
