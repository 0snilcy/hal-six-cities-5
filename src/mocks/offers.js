// const getRandomNumber = (number) => getCountRandom(0, number);
const getRandomBoolean = () => Boolean(Math.round(Math.random()));

const getCountRandom = (min, max) =>
  Math.floor(getCountRandomDecimal(max, min));

const getCountRandomDecimal = (min, max) =>
  Number((Math.random() * (max - min) + min).toFixed(1));

export default {
  isPremium: getRandomBoolean(),
  name: [
    `Beautiful & luxurious apartment at great location`,
    `Wood and stone place`,
    `Nice, cozy, warm big bed apartment`,
    `Canal View Prinsengracht`,
  ][getCountRandom(0, 3)],
  rating: getCountRandomDecimal(2, 5),
  classRoom: [`apartments`, `private room`, `studio`, `loft`][
    getCountRandom(0, 3)
  ],
  bedrooms: getCountRandom(1, 4),
  adults: getCountRandom(1, 5),
  price: Math.round(getCountRandom(40, 200) / 10) * 10,
  tariff: [`night`, `light`][getCountRandom(0, 1)],
  conveniences: [
    `Wi-Fi`,
    `Washing machine`,
    `Towels`,
    `Heating`,
    `Coffee machine`,
    `Baby seat`,
    `Kitchen`,
    `Dishwasher`,
    `Cabel TV`,
    `Fridge`,
  ].slice(getCountRandom(0, 4), getCountRandom(4, 9)),
};
