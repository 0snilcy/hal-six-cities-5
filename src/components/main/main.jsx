import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import PlacesList from '../places-list/places-list';
import Map from '../map/map';
import { propsOffers } from '../../props/props';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { ActionCreator } from '../store/action';
import { createArrOffers } from '../../utils/utils';
import CityList from '../city-list/city-list';
import cl from 'classnames';
import { city } from '../../mocks/data';

export const arrOffers = createArrOffers(city.length);

class Main extends PureComponent {
  render() {
    console.log(this.props);
    const {
      countOffersRent,
      offers,
      history,
      changeCity,
      loadingPlacesList,
      activeCity,
      activeOffers,
    } = this.props;
    return (
      <div className="page page--gray page--main">
        <header className="header">
          <div className="container">
            <div className="header__wrapper">
              <div className="header__left">
                <a className="header__logo-link header__logo-link--active">
                  <img
                    className="header__logo"
                    src="img/logo.svg"
                    alt="6 cities logo"
                    width="81"
                    height="41"
                  />
                </a>
              </div>
              <nav className="header__nav">
                <ul className="header__nav-list">
                  <li className="header__nav-item user">
                    <Link
                      to="/favorites"
                      className="header__nav-link header__nav-link--profile"
                    >
                      <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                      <span className="header__user-name user__name">
                        Oliver.conner@gmail.com
                      </span>
                    </Link>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </header>
        <main className="page__main page__main--index">
          <h1 className="visually-hidden">Cities</h1>
          <div className="tabs">
            <section className="locations container">
              {/* <CityList /> */}
              <ul
                className="locations__list tabs__list"
                onClick={({ target }) => {
                  changeCity(target.textContent);
                }}
              >
                {city.map((el, id) => {
                  return (
                    <li className="locations__item" key={id}>
                      <a
                        className={cl('locations__item-link tabs__item', {
                          'tabs__item--active': el === activeCity,
                        })}
                        href="#"
                        onClick={() => loadingPlacesList(arrOffers[id])}
                      >
                        <span>{el}</span>
                      </a>
                    </li>
                  );
                })}
              </ul>
            </section>
          </div>
          <div className="cities">
            <div className="cities__places-container container">
              <section className="cities__places places">
                <h2 className="visually-hidden">Places</h2>
                <b className="places__found">
                  {countOffersRent} places to stay in Amsterdam
                </b>
                <form className="places__sorting" action="#" method="get">
                  <span className="places__sorting-caption">Sort by</span>
                  <span className="places__sorting-type" tabIndex="0">
                    Popular
                    <svg className="places__sorting-arrow" width="7" height="4">
                      <use xlinkHref="#icon-arrow-select"></use>
                    </svg>
                  </span>
                  <ul className="places__options places__options--custom places__options--opened">
                    <li
                      className="places__option places__option--active"
                      tabIndex="0"
                    >
                      Popular
                    </li>
                    <li className="places__option" tabIndex="0">
                      Price: low to high
                    </li>
                    <li className="places__option" tabIndex="0">
                      Price: high to low
                    </li>
                    <li className="places__option" tabIndex="0">
                      Top rated first
                    </li>
                  </ul>
                </form>
                <PlacesList offers={activeOffers} />
              </section>
              <div className="cities__right-section">
                <section className="cities__map map">
                  <Map offers={activeOffers} />
                </section>
              </div>
            </div>
          </div>
        </main>
      </div>
    );
  }
}

Main.propTypes = {
  countOffersRent: PropTypes.number.isRequired,
  offers: PropTypes.arrayOf(PropTypes.shape(propsOffers)),
};

const mapStateToProps = (state) => ({
  activeCity: state.city,
  activeOffers: state.offers,
});
const mapDispatchToProps = (dispatch) => ({
  changeCity: (payload) => dispatch(ActionCreator.changeCity(payload)),
  loadingPlacesList: (payload) =>
    dispatch(ActionCreator.loadingPlacesList(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Main);
