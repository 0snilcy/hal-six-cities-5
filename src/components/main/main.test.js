import React from 'react';
import renderer from 'react-test-renderer';
import { getOffer } from '../../mocks/data';
import { Main } from '../main/main';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

jest.mock('../map/map.jsx', () => 'Map');

const middlewares = [];
const mockStore = configureStore(middlewares);

const offers = new Array(10).fill('').map(getOffer);
const activeOffer = offers.slice(1, 4);
const userData = {
  email: 'zubov@infonica.ru',
  avatarUrl:
    'https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg',
};

const initialState = {
  PROCESS: {
    activeCity: `Amsterdam`,
  },
};
const store = mockStore(initialState);

describe('<Main /> render', () => {
  it('testMain', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Main
              offers={offers}
              activeOffer={activeOffer}
              userData={userData}
            />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
