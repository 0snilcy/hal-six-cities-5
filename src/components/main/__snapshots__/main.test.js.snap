// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> render testMain 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                  style={
                    Object {
                      "backgroundImage": "url(https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg)",
                    }
                  }
                />
                <span
                  className="header__user-name user__name"
                >
                  zubov@infonica.ru
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
          onClick={[Function]}
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Paris
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Cologne
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Brussels
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
            >
              <span>
                Amsterdam
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Hamburg
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Dusseldorf
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            3
             places to stay in 
            Amsterdam
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
            >
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom places__options--closed"
            >
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="places__list cities__places-list tabs__content"
          >
            <article
              className="place-card cities__place-card"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="place-card__mark"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  href="/offer/10855531487849812"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/room-small.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      â‚¬
                      130
                    </b>
                    <span
                      className="place-card__price-text"
                    />
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "1480%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer/10855531487849812"
                    onClick={[Function]}
                  >
                    Wood and stone place
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  studio
                </p>
              </div>
            </article>
            <article
              className="place-card cities__place-card"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="place-card__mark"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  href="/offer/4543963331385126"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/room-small.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      â‚¬
                      60
                    </b>
                    <span
                      className="place-card__price-text"
                    />
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "1639.9999999999995%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer/4543963331385126"
                    onClick={[Function]}
                  >
                    Nice, cozy, warm big bed apartment
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  studio
                </p>
              </div>
            </article>
            <article
              className="place-card cities__place-card"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="place-card__mark"
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  href="/offer/88986002318173"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/room-small.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      â‚¬
                      100
                    </b>
                    <span
                      className="place-card__price-text"
                    />
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "1519.9999999999995%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer/88986002318173"
                    onClick={[Function]}
                  >
                    Wood and stone place
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  private room
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <Map
              offers={
                Array [
                  Object {
                    "adults": 2,
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Brussels",
                    },
                    "description": "Great hotel. Good location. Everything is clean and cozy. A variety of breakfasts. I thank the hotel staff. I would like to note that everyone is very friendly and polite, so an atmosphere of good rest and excellent mood is created.",
                    "goods": Array [
                      "Washing machine",
                      "Towels",
                      "Heating",
                      "Coffee machine",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 3,
                      "name": "Angelina",
                    },
                    "id": 10855531487849812,
                    "images": Array [
                      "img/apartment-01.jpg",
                      "img/apartment-02.jpg",
                      "img/apartment-03.jpg",
                    ],
                    "isFavorite": true,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 52.3814938496678,
                      "longitude": 4.629877537499948,
                    },
                    "previewImage": "img/room-small.jpg",
                    "price": 130,
                    "rating": 74,
                    "tariff": "night",
                    "title": "Wood and stone place",
                    "type": "studio",
                  },
                  Object {
                    "adults": 4,
                    "bedrooms": 2,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Cologne",
                    },
                    "description": "Very recommended: comfortable location, incredibly friendly staff, very tasty breakfasts, cleanliness, comfort. A decent hotel! You're great, so keep it! Thanks for the hospitality, be sure to return to you.",
                    "goods": Array [
                      "Towels",
                      "Heating",
                      "Coffee machine",
                      "Baby seat",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 3,
                      "name": "Angelina",
                    },
                    "id": 4543963331385126,
                    "images": Array [
                      "img/apartment-01.jpg",
                      "img/apartment-02.jpg",
                      "img/apartment-03.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 52.3714938496678,
                      "longitude": 4.690877537499948,
                    },
                    "previewImage": "img/room-small.jpg",
                    "price": 60,
                    "rating": 81.99999999999999,
                    "tariff": "night",
                    "title": "Nice, cozy, warm big bed apartment",
                    "type": "studio",
                  },
                  Object {
                    "adults": 2,
                    "bedrooms": 1,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Hamburg",
                    },
                    "description": "Very recommended: comfortable location, incredibly friendly staff, very tasty breakfasts, cleanliness, comfort. A decent hotel! You're great, so keep it! Thanks for the hospitality, be sure to return to you.",
                    "goods": Array [
                      "Heating",
                      "Coffee machine",
                      "Baby seat",
                      "Kitchen",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 3,
                      "name": "Angelina",
                    },
                    "id": 88986002318173,
                    "images": Array [
                      "img/apartment-01.jpg",
                      "img/apartment-02.jpg",
                      "img/apartment-03.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": true,
                    "location": Object {
                      "latitude": 52.3514938496678,
                      "longitude": 4.612877537499948,
                    },
                    "previewImage": "img/room-small.jpg",
                    "price": 100,
                    "rating": 75.99999999999999,
                    "tariff": "night",
                    "title": "Wood and stone place",
                    "type": "private room",
                  },
                ]
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
